Chessington.GameEngine/Pieces/Knight.cs

```diff
@@ -10,7 +10,36 @@ public Knight(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var availableMoves = new List<Square>();
+
+            for (var i = 0; i <= 7; i++)
+            {
+                var row = KnightPosition(board.FindPiece(this), i).Row;
+                var col = KnightPosition(board.FindPiece(this), i).Col;
+                if (CheckPosition(row, col))
+                {
+                    availableMoves.Add(Square.At(row,col));
+                }
+            }
+
+            return availableMoves;
+        }
+        
+        private bool CheckPosition(int row, int col)
+        {
+            return row < GameSettings.BoardSize && row >= 0 && col < GameSettings.BoardSize && col >= 0;
+        }
+
+        private Square KnightPosition(Square initialPosition, int direction)
```

Comment:
not too bad, but seems a bit complicated.
you can just return list of 8 squares manually. And then do possibleSquares.Where(x => CheckPosition(x))